packages/playwright-core/src/server/chromium/crServiceWorker.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Worker } from '../page';
+import type { CRBrowserContext } from './crBrowser';
+import type { CRSession } from './crConnection';
+import { CRExecutionContext } from './crExecutionContext';
+import { CRNetworkManager } from './crNetworkManager';
+import * as network from '../network';
+import { BrowserContext } from '../browserContext';
+
+export class CRServiceWorker extends Worker {
+  readonly _browserContext: CRBrowserContext;
+  readonly _networkManager: CRNetworkManager;
+
+  constructor(browserContext: CRBrowserContext, session: CRSession, url: string) {
+    super(browserContext, url);
+    this._browserContext = browserContext;
+    this._networkManager = new CRNetworkManager(session, null, this, null);
+    session.once('Runtime.executionContextCreated', event => {
+      this._createExecutionContext(new CRExecutionContext(session, event.context));
+    });
+
+    // These might fail if the target is closed before we receive all execution contexts.
+    this._networkManager.initialize().catch(() => {});
```

Comment:
We'll need to think about how to opt people in/out. Doing this unconditionally would risk old code triggering an error by calling `request.frame()`.